### PR TITLE
guzzle auto-instrumentation

### PR DESCRIPTION
**Pull request by brettmc**
Fixes: https://github.com/open-telemetry/opentelemetry-php/issues/1204

### CODE REVIEW

File: src/Instrumentation/Guzzle/composer.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  "readme": "./README.md",
+  "license": "Apache-2.0",
+  "minimum-stability": "dev",
+  "prefer-stable": true,
+  "require": {
+    "php": "^8.0",
+    "ext-opentelemetry": "*",
+    "ext-reflection": "*",
+    "open-telemetry/api": "^1.0",
+    "open-telemetry/sem-conv": "^1.22",
```
[the remark this answers is not in the document]
Comment (brettmc) [reply src/Instrumentation/Guzzle/composer.json]: I think it's a good idea to do a separate PR to update everything to 1.23

**Comment by bobstrecansky** [comment]
Some uncaught Phan exceptions; this expected?

**Comment by brettmc** [comment]
> Some uncaught Phan exceptions; this expected?

No. I added 8.3 to the build matrix and most jobs seem unhappy with it. I'll tackle that as a separate PR...

**Review by bobstrecansky — APPROVED** [review]

**Comment by brettmc** [comment]
https://packagist.org/packages/open-telemetry/opentelemetry-auto-guzzle

**Pull request merged**